### PR TITLE
[vscoq2] cleanup options data type and logics

### PR DESCRIPTION
**Pull request by gares**

### CODE REVIEW

**Review by maximedenes — DISMISSED** [review]
After discussion: maybe better to put a ref in execution manager and then switch to an (explicit) global state in exec manager, instead of per-doc.

**Comment by gares** [comment]
done

**Comment by gares** [comment]
Damn, I need a green light to merge. @maximedenes would you unblock this swiftly?

**Pull request merged**
